### PR TITLE
Added TMDataGrid

### LINKED ISSUE #147: Table | Basic setup
- Table class have to be created
 - All the necessary properties are exposed (ItemSource, ItemSelected) 
 - Columns Property is exposed (Column size | Equal, Percentage, Auto)

### LINKED ISSUE #148: Table | Column Type | TextColumn -> It can have title
- PropertyName and HeaderName are exposed
 - Always left-aligned

### LINKED ISSUE #149: Table | Column Type | BooleanColumn -> It will have a switch

### LINKED ISSUE #150: Table | Column Type | ImageColumn -> It will have a image
- HeaderName, ImageSource are exposed
- Sorting is disabled by default
- Always center aligned

### LINKED ISSUE #151: Table | Column Type | TemplateColumn -> User have full control over the content

### LINKED ISSUE #152: Table | Sorting
- By default, the items are displayed in the order given by the user
 - When the header is tapped, we sort the items either asc/desc
 - Sorting is applicable for all the colulmns and it is false by default

### LINKED ISSUE #154: Table | Selection
- Seleting a row, marks the items in bluecolor
 - A property SelectionMode should be exposed with None, Single, Multiple options
- SelectedItem properties should be exposed and will have the selected row items
- SelectionChangedCommand should be exposed

### PR DESCRIPTION
**Pull request by BAL4-R**
Closes #152 , closes #154 , closes #147 , closes #148 , closes #149 , closes #150, closes #151 
Screenshots:
 

WhatsApp Image 2023-07-31 at 1 59 24 PM (1)
WhatsApp Image 2023-07-31 at 1 59 24 PM

### CODE REVIEW

File: DemoApp/DemoApp/MainPage.xaml.cs [hunk 1]
```diff
@@ -1,80 +1,84 @@
...
+                        Navigation.PushAsync(new TMSpinnerPage());
+                        break;
+                    case "SegmentedControl":
+                        Navigation.PushAsync(new TMSegmentedControlPage());
+                        break;
+                    case "ListView":
+                        Navigation.PushAsync(new TMListViewPage());
+                        break;
+                    case "DataGrid":
+                        Navigation.PushAsync(new TableView());
```
Comment (rthanga1) [inline DemoApp/DemoApp/MainPage.xaml.cs:73]: Can you rename this page name to `TableViewPage`

File: DemoApp/DemoApp/ViewModels/TableViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    [NotifyPropertyChangedFor(nameof(SelectionMode))]
+    private int _selectionModeRadioOption = 0;
+    partial void OnSelectionModeRadioOptionChanging(int value)
+    {
+        SelectionMode = (SelectionMode)value;
+    }
+
+    public TableViewModel()
+    {
+        var user1 = new User()
```
Comment (rthanga1) [inline DemoApp/DemoApp/ViewModels/TableViewModel.cs:27]: Can you move the data set to a json file and read it from there? In that way, if we want to add more user data, we can just edit the json file. Refer 'UserData.json' in the Raw folder.

File: DemoApp/DemoApp/Views/TableView.xaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:modus="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
+             x:Class="DemoApp.TableView"
+             Title="DataGrid">
+  <ScrollView>
```
Comment (rthanga1) [inline DemoApp/DemoApp/Views/TableView.xaml:7]: Please remove the root ScrollView. That is not needed here

File: DemoApp/DemoApp/Views/TableView.xaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+              <DataTemplate>
+                <StackLayout VerticalOptions="Center">
+                  <Label Text="{Binding ID}"
+                         Padding="16,0" />
+                </StackLayout>
+              </DataTemplate>
+            </modus:DataGridColumn.CellTemplate>
+          </modus:DataGridColumn>
+          <!-- Boolean column -->
+          <modus:BooleanColumn Title="Qualified"
```
Comment (rthanga1) [inline DemoApp/DemoApp/Views/TableView.xaml:53]: Can you also add `*` and a `fixed width` value? I think we can have fixed value fo the boolean column and the remaining space should be taken by the name column

File: Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/BooleanColumn.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+namespace Trimble.Modus.Components;
+
+public partial class BooleanColumn
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/BooleanColumn.xaml.cs:3]: Add the parent class DataGridColumn. Easy to see that it is a subclass without looking at the xaml file
`public partial class BooleanColumn : DataGridColumn`

File: DemoApp/DemoApp/ViewModels/TableViewModel.cs (hunk 2 above)
Comment (rthanga1) [reply DemoApp/DemoApp/ViewModels/TableViewModel.cs:27]: And refer the data from the UserData.json and use the usernames and logo images from there

File: Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/TextColumn.xaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                      xmlns:modus="clr-namespace:Trimble.Modus.Components"
+                      x:Class="Trimble.Modus.Components.TextColumn">
+  <modus:DataGridColumn.CellTemplate>
+    <DataTemplate>
+      <Grid RowSpacing="2">
+        <Grid.RowDefinitions>
+          <RowDefinition Height="*" />
+          <RowDefinition Height="Auto" />
+        </Grid.RowDefinitions>
+        <Label Grid.Row="0"
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/TextColumn.xaml:13]: I don't see any bindings or x:Name or anything here. Can you explain how this works?

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      <CollectionView.ItemTemplate>
+        <DataTemplate>
+          <local:DataGridRow MinimumHeightRequest="48"
+                             Padding="0,0,0,1">
+            <VisualStateManager.VisualStateGroups>
+              <VisualStateGroup Name="CommonStates">
+                <VisualState Name="Normal" />
+                <VisualState Name="Selected">
+                  <VisualState.Setters>
+                    <Setter Property="BackgroundColor"
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml:57]: The selection doesn't seem to work.
Are you setting the BackgroundColor to Transparent when Selected? Can you confirm if this is really working?

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,727 @@
...
+    /// </summary>
+    /// <param name="item">Item to scroll</param>
+    /// <param name="position">Position of the row in screen</param>
+    /// <param name="animated">animated</param>
+    public void ScrollTo(object item, ScrollToPosition position, bool animated = true) => _collectionView.ScrollTo(item, position: position, animate: animated);
+
+    #endregion Methods
+
+    #region Bindable properties
+    public static readonly BindableProperty ReadOnlyListProperty = BindableProperty.Create(nameof(ReadOnlyList), typeof(List<int>), typeof(DataGrid), defaultValueCreator: _ => new List<int>());
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:223]: What is the user of this ReadOnlyList property? I don't see it is used anywhere other than the component itself. Do we need a BindableProperty for this?

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,727 @@
...
+    public static readonly BindableProperty SelectionModeProperty = BindableProperty.Create(nameof(SelectionMode), typeof(SelectionMode), typeof(DataGrid), Microsoft.Maui.Controls.SelectionMode.None, BindingMode.TwoWay, propertyChanged: OnSelectionModeChanged);
+    public static readonly BindableProperty SortIconProperty = BindableProperty.Create(nameof(SortIcon), typeof(Polygon), typeof(DataGrid));
+    public static readonly BindableProperty NoDataViewProperty = BindableProperty.Create(nameof(NoDataView), typeof(View), typeof(DataGrid), propertyChanged: OnNoDataViewChanged);
+    public static readonly BindableProperty ShowDividerProperty = BindableProperty.Create(nameof(ShowDivider), typeof(bool), typeof(DataGrid), true, propertyChanged: OnShowDividerPropertyChanged);
+    public static readonly BindableProperty SelectedItemsProperty = BindableProperty.Create(nameof(SelectedItems), typeof(IList<object>), typeof(DataGrid), null, BindingMode.TwoWay);
+    public static readonly BindableProperty SelectedItemProperty = BindableProperty.Create(nameof(SelectedItem), typeof(object), typeof(DataGrid), null, BindingMode.TwoWay,
+        propertyChanged: (bindable, _, newValue) =>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:234]: Can you move these to separate methods?

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,727 @@
...
+    public static readonly BindableProperty ColumnsProperty = BindableProperty.Create(nameof(Columns), typeof(ObservableCollection<DataGridColumn>), typeof(DataGrid), propertyChanged: OnColumnChanged, defaultValueCreator: _ => new ObservableCollection<DataGridColumn>());
+    public static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(DataGrid), null, propertyChanged: OnItemsSourceChanged);
+    public static readonly BindableProperty IsSortableProperty = BindableProperty.Create(nameof(IsSortable), typeof(bool), typeof(DataGrid), false);
+    public static readonly BindableProperty SelectionModeProperty = BindableProperty.Create(nameof(SelectionMode), typeof(SelectionMode), typeof(DataGrid), Microsoft.Maui.Controls.SelectionMode.None, BindingMode.TwoWay, propertyChanged: OnSelectionModeChanged);
+    public static readonly BindableProperty SortIconProperty = BindableProperty.Create(nameof(SortIcon), typeof(Polygon), typeof(DataGrid));
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:229]: Don't have to expose Sort icon

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,727 @@
...
+    /// <param name="item">Item to scroll</param>
+    /// <param name="position">Position of the row in screen</param>
+    /// <param name="animated">animated</param>
+    public void ScrollTo(object item, ScrollToPosition position, bool animated = true) => _collectionView.ScrollTo(item, position: position, animate: animated);
+
+    #endregion Methods
+
+    #region Bindable properties
+    public static readonly BindableProperty ReadOnlyListProperty = BindableProperty.Create(nameof(ReadOnlyList), typeof(List<int>), typeof(DataGrid), defaultValueCreator: _ => new List<int>());
+    public static readonly BindableProperty ItemSizingStrategyProperty = BindableProperty.Create(nameof(ItemSizingStrategy), typeof(ItemSizingStrategy), typeof(DataGrid), DeviceInfo.Platform == DevicePlatform.Android ? ItemSizingStrategy.MeasureAllItems : ItemSizingStrategy.MeasureFirstItem);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:224]: Need of ItemSizingStrategyProperty?

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 12]
```diff
@@ -0,0 +1,727 @@
...
+    public static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(DataGrid), null, propertyChanged: OnItemsSourceChanged);
+    public static readonly BindableProperty IsSortableProperty = BindableProperty.Create(nameof(IsSortable), typeof(bool), typeof(DataGrid), false);
+    public static readonly BindableProperty SelectionModeProperty = BindableProperty.Create(nameof(SelectionMode), typeof(SelectionMode), typeof(DataGrid), Microsoft.Maui.Controls.SelectionMode.None, BindingMode.TwoWay, propertyChanged: OnSelectionModeChanged);
+    public static readonly BindableProperty SortIconProperty = BindableProperty.Create(nameof(SortIcon), typeof(Polygon), typeof(DataGrid));
+    public static readonly BindableProperty NoDataViewProperty = BindableProperty.Create(nameof(NoDataView), typeof(View), typeof(DataGrid), propertyChanged: OnNoDataViewChanged);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:230]: Rename this to`EmptyView`. That's the standard MAUI uses for all other layouts

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 13]
```diff
@@ -0,0 +1,727 @@
...
+            if (self.SelectionMode == SelectionMode.None)
+            {
+                return null;
+            }
+
+            return value;
+        }
+    );
+    public static readonly BindableProperty SortedColumnIndexProperty = BindableProperty.Create(nameof(SortedColumnIndex), typeof(SortData), typeof(DataGrid), null, BindingMode.TwoWay,
+        (bindable, value) =>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:254]: Same here, see if you can move these to separate method

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 14]
```diff
@@ -0,0 +1,727 @@
...
+    /// </summary>
+    public IList<object> SelectedItems
+    {
+        get => (IList<object>)GetValue(SelectedItemsProperty);
+        set => SetValue(SelectedItemsProperty, value);
+    }
+    /// <summary>
+    /// Column index and sorting order for the DataGrid
+    /// </summary>
+    public SortData SortedColumnIndex
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:498]: The name and the type doesn't match. SortedColumnIndex feels like it is an int but it is not.

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 15]
```diff
@@ -0,0 +1,727 @@
...
+    /// </summary>
+    internal Color BorderColor = ResourcesDictionary.ColorsDictionary(ColorsConstants.TableDividerColor);
+    /// <summary>
+    /// Border thickness for header &amp; each cell
+    /// </summary>
+    internal Thickness BorderThickness = new(0, 0, 0, 1);
+    /// <summary>
+    /// Readonly row color
+    /// </summary>
+    internal Color ReadOnlyRowColor = ResourcesDictionary.ColorsDictionary(ColorsConstants.NeutralGray);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:45]: Where is this used?

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 16]
```diff
@@ -0,0 +1,727 @@
...
+    /// Reload the table
+    /// </summary>
+    internal void Reload()
+    {
+        InitHeaderView();
+        _collectionView.BackgroundColor = BorderColor;
+
+        if (_internalItems is not null)
+        {
+            InternalItems = new List<object>(_internalItems);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:592]: Not sure if I get this right. _internalItems gets updated whenever the setter of InternalItems is updated, right? So why is this needed?

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGridColumn.cs [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+    public static readonly BindableProperty CellTemplateProperty = BindableProperty.Create(nameof(CellTemplate), typeof(DataTemplate), typeof(DataGridColumn));
+    public static readonly BindableProperty LineBreakModeProperty = BindableProperty.Create(nameof(LineBreakMode), typeof(LineBreakMode), typeof(DataGridColumn), LineBreakMode.WordWrap);
+    public static readonly BindableProperty HorizontalContentAlignmentProperty = BindableProperty.Create(nameof(HorizontalContentAlignment), typeof(LayoutOptions), typeof(DataGridColumn), LayoutOptions.Center);
+    public static readonly BindableProperty VerticalContentAlignmentProperty = BindableProperty.Create(nameof(VerticalContentAlignment), typeof(LayoutOptions), typeof(DataGridColumn), LayoutOptions.Center);
+    public static readonly BindableProperty SortingEnabledProperty = BindableProperty.Create(nameof(SortingEnabled), typeof(bool), typeof(DataGridColumn), true);
+    public static readonly BindableProperty FormattedTitleProperty = BindableProperty.Create(nameof(FormattedTitle), typeof(FormattedString), typeof(DataGridColumn), propertyChanged: (b, _, n) => ((DataGridColumn)b).HeaderLabel.FormattedText = (FormattedString)n);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/DataGridColumn.cs:55]: Is the FormattedTitle used anywhere?

File: Trimble.Modus.Components/Controls/TMDataGrid/Maui.DataGrid.csproj [hunk 18]
```diff
@@ -0,0 +1,40 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMDataGrid/Maui.DataGrid.csproj:1]: Why is this here? The DataGrid is not added as a separate project right?

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/Model/User.cs [hunk 19]
```diff
@@ -0,0 +1,12 @@
+namespace DemoApp.Model;
+
+public class User
+{
```
[the remark this answers is not in the document]
Comment (rthanga1) [reply DemoApp/DemoApp/Model/User.cs:4]: I concur. But we needed a boolean property to show how the boolean column looks like. Can you a Qualified property here and add some sample boolean values to the userdata.json file? And reuse the loadData method from Sam's listview demo page?

File: DemoApp/DemoApp/MainPage.xaml.cs (hunk 1 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/MainPage.xaml.cs]: Done

File: DemoApp/DemoApp/ViewModels/TableViewModel.cs (hunk 2 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/ViewModels/TableViewModel.cs]: Done

File: DemoApp/DemoApp/Views/TableView.xaml (hunk 3 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/Views/TableView.xaml]: ScrollView is needed since the table has more items now

File: DemoApp/DemoApp/Views/TableView.xaml (hunk 4 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/Views/TableView.xaml]: Done

File: Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/BooleanColumn.xaml.cs (hunk 5 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/BooleanColumn.xaml.cs]: Done

File: Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/TextColumn.xaml (hunk 6 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/ColumnTypes/TextColumn.xaml:13]: The binding is set in `DataGridRow.CreateCell()` method.

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml (hunk 7 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml:55]: In CollectionView when an item is selected it will have an yellow background color, this VisualState manager is set to hide that selection color. We set our custom selection color is code behind

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 8 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:223]: Guidelines has a "Read Only row" concept, which has a different background color and cannot be selected. 'ReadOnlyList' takes a list of row index as input.
'

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 9 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs]: Done

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 11 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:224]: This property is available in MAUI CollectionView, hence added it here

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 12 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs]: Done

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 13 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs]: Done

**Review by ElishaSamPeterPrabhu — APPROVED** [review]

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 14 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs]: Renamed

File: DemoApp/DemoApp/Model/User.cs (hunk 19 above)
[the remark this answers is not in the document]
Comment (BAL4-R) [reply DemoApp/DemoApp/Model/User.cs]: Done

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+        <Setter Property="Padding"
+                Value="16,0" />
+      </Style>
+    </ResourceDictionary>
+  </ContentView.Resources>
+  <Grid CompressedLayout.IsHeadless="True">
+    <Grid.RowDefinitions>
+      <RowDefinition Height="Auto" />
+      <RowDefinition Height="*" />
+      <RowDefinition Height="Auto" />
```
[the remark this answers is not in the document]
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml]: Not used, removed

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs [hunk 21]
```diff
@@ -0,0 +1,727 @@
...
+                self.SortColumn(sortData);
+            }
+        });
+    #endregion Bindable properties
+
+    #region Property changed handlers
+    /// <summary>
+    /// Triggered when column is changed
+    /// </summary>
+    private static void OnColumnChanged(BindableObject bindable, object oldValue, object newValue)
```
[the remark this answers is not in the document]
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs]: Done

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 16 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs:581]: We are assigning the items in `_internalItems` to `InternalItems` so that it will be considered as new list of items and the rows will reloaded.

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGridColumn.cs (hunk 17 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGridColumn.cs:55]: Its not used by us, its a property to set formatted header title for columns.

File: Trimble.Modus.Components/Controls/TMDataGrid/Maui.DataGrid.csproj (hunk 18 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/Maui.DataGrid.csproj]: This is not needed, removed

File: DemoApp/DemoApp/Views/TableView.xaml (hunk 3 above)
Comment (rthanga1) [reply DemoApp/DemoApp/Views/TableView.xaml]: Hope this is addressed now

Comment (rthanga1) [reply DemoApp/DemoApp/Views/TableView.xaml]: It looks like the scrollview is still there. Like we discussed, remove the ScrollView and VeriticalStackLayout and have just a StackLayout

File: DemoApp/DemoApp/DemoApp.csproj [hunk 22]
```diff
@@ -84,6 +84,9 @@
 	</ItemGroup>
 
 	<ItemGroup>
+	  <Compile Update="Views\TableViewPage.xaml.cs">
```
Comment (rthanga1) [inline DemoApp/DemoApp/DemoApp.csproj:87]: Performance is very poor when we have too many records - like 100.

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGridColumn.cs (hunk 17 above)
Comment (rthanga1) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGridColumn.cs]: I guess this is removed now?

Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGridColumn.cs]: This property is removed

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 15 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs]: Not needed, removed

File: Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs (hunk 11 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs]: This property is now removed

File: DemoApp/DemoApp/Views/TableView.xaml (hunk 3 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/Views/TableView.xaml]: ScrollView and VerticalStackLayout are removed and replaced with Grid

File: DemoApp/DemoApp/DemoApp.csproj (hunk 22 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/DemoApp.csproj:87]: Created a new ticket for this issue, #192

**Review by rthanga1 — APPROVED** [review]
